modules/ROOT/pages/6.4.1-release-notes.adoc

```diff
@@ -0,0 +1,1138 @@
+
+=== Non breaking space is added after backspace below text pattern heading
+//#TINY-9603
+
+A recently discovered issue was identified in previous versions of {productname}, that affected the text formatting feature for **Text Pattern**. When a user would apply a text pattern heading such as `# Title`, followed by pressing "enter", the issue would the occur when the backspace key was pressed. As a result, a non-breaking space was being inserted to the beginning of the line instead. This unintended addition of a Non-breaking space caused formatting inconsistencies and errors.
+
+In {productname} 6.4.1, pressing backspace after adding a Text Pattern, now formats the content as expected.
+
+=== Incorrect event target for dragend and drop events when dragging and dropping noneditable elements within editor
+//#TINY-9599
+
+In previous versions of {productname}, when dragging and dropping `contenteditable="false"` elements within the editor, the target of the `dragstart`, `dragend`, and `drop` events would be the editor instance itself, rather than the expected DOM target.
+
+In {productname} 6.4.1, the events are now simulated based on the mouse event that produced them, meaning that the target and other properties, such as mouse coordinates, are now properly produced for the simulated `dragstart`, `dragend`, and `drop` events.
+
+=== Dialog getting cut offscreen when opened from toolbar button in Inline Mode
+//#TINY-9588
+
+In previous versions of {productname}, inline dialogs would always open to the South West of the button used to open it from the toolbar, or South East with RTL (Right-to-Left) directionality. However, an issue was found where, when the toolbar was shorter than the inline dialog, inline dialogs would appear partially off screen when opened.
+
+To address this issue, inline dialogs now determine a more appropriate direction to open, that is, any direction which will not cause the dialog to appear offscreen.
+
+=== Checkmark icons are added to every color in color swatch
+//#TINY-9567
+
+In previous versions of {productname}, when using certain configurations, the color swatch feature could add a checkmark icon to every color, when the checkmark should only be displayed on a color when one is enabled.
+
+In {productname} 6.4.1, the selector for applying the checkmark is updated so the checkmark will only be added and displayed for enabled colors.
+
+=== The colour picker reset to default when formatting pre element with addtional format options
+//#TINY-9342
+
+In previous versions of {productname}, formatting elements with a `pre-formatted` tag, and applying a background color from the color swatch would cause formatting issues when additional formatting options such as bold and italics were applied to the `pre-formatted` element. When these formatting conditions were combined, the color picker would reset the users defined color value, which would triggered the color picker to reset this value to default "000000".
```

Comment:
```suggestion
In previous versions of {productname}, formatting elements with a `pre-formatted` tag, and applying a background color from the color swatch would cause formatting issues when additional formatting options such as bold and italics were applied to the `pre-formatted` element. When these formatting conditions were combined, the color picker would reset the users defined color value, which would trigger the color picker to reset this value to default "000000".
```